src/components/Main.js

```diff
@@ -18,130 +18,134 @@ import '../styles/Main.scss';
-        if (this.state.viewMode === CODE_AND_OUTPUT) {
-          this.setState({ viewMode: CODE_ONLY });
-        }
+  useEffect(() => {
+    if (screenWidth <= EDITOR_WIDTH_BREAKPOINT) {
+      if (viewMode === CODE_AND_OUTPUT) {
+        setViewMode(CODE_ONLY);
       }
     }
-  }
+  }, [screenWidth]);
 
-  onThemeChange = () => {
-    const newTheme = this.props.theme === 'dark' ? 'light' : 'dark';
+  const onThemeChange = () => {
+    const newTheme = theme === 'dark' ? 'light' : 'dark';
     cookies.setThemeCookie(newTheme);
-    this.props.setTheme(newTheme);
+    setTheme(newTheme);
   };
 
-  resetSaveText = () => {
-    this.setState({
-      saveText: 'Save',
-    });
+  const resetSaveText = () => {
+    setSaveText('Save');
   };
 
-  handleSave = () => {
-    if (!this.props.dirty) return; // Don't save if not dirty (unedited)
-    this.setState({
-      saveText: 'Saving...',
-    });
+  const handleSave = () => {
+    if (!dirty) return; // Don't save if not dirty (unedited)
+    setSaveText('Saving...');
 
     const programToUpdate = {};
-
-    programToUpdate[this.props.mostRecentProgram] = {
-      code: this.props.code,
+    programToUpdate[mostRecentProgram] = {
+      code,
     };
 
-    fetch.updatePrograms(this.props.uid, programToUpdate).then(() => {
-      this.setState({
-        saveText: 'Saved!',
-      });
+    fetch.updatePrograms(uid, programToUpdate).then(() => {
+      setSaveText('Saved!');
 
-      setTimeout(this.resetSaveText, 3000);
+      setTimeout(resetSaveText, 3000);
     });
-    this.props.cleanCode(this.props.mostRecentProgram); // Set code's "dirty" state to false
-  };
-
-  renderContent = () => {
-    switch (this.props.contentType) {
-      case "editor":
-        return this.renderEditor();
-      case "classes":
-        return <ClassesPageContainer />;
-      case "classPage":
-        return <ClassPageContainer />;
-      case "sketches":
-      default:
-        return <SketchesPageContainer />;
-    }
+    cleanCode(mostRecentProgram); // Set code's "dirty" state to false
   };
 
-  renderEditor = () => (
+  const renderEditor = () => (
     <EditorAndOutput
       // view mode
-      viewMode={this.state.viewMode}
-      updateViewMode={(viewMode) => this.setState({ viewMode })}
+      viewMode={viewMode}
+      updateViewMode={(mode) => setViewMode(mode)}
       // theme
-      theme={this.props.theme}
+      theme={theme}
       // sizing
-      left={this.props.left}
-      screenWidth={this.props.screenWidth}
-      screenHeight={this.props.screenHeight}
+      left={left}
+      screenWidth={screenWidth}
+      screenHeight={screenHeight}
       // view only trigger
       viewOnly={false}
       // pane
-      panelOpen={this.props.panelOpen}
-      pane1Style={this.state.pane1Style}
-      changePane1Style={(newStyle) => this.setState(newStyle)}
+      panelOpen={panelOpen}
+      pane1Style={pane1Style}
+      changePane1Style={(newStyle) => setPane1Style(newStyle)}
```

Comment:
Couple things here:
First, this can probably be simplified to just `setPane1Style` instead of a function which does the same thing
Second, this is a functionality change; you need to update the input to this function in `EditorAndOutput.js` as well